art/attacks/evasion/projected_gradient_descent/projected_gradient_descent_pytorch.py

```diff
@@ -85,13 +85,7 @@ def __init__(
         :param batch_size: Size of the batch on which adversarial samples are generated.
         :param verbose: Show progress bars.
         """
-        if (
-            hasattr(estimator, "preprocessing")
-            and (estimator.preprocessing is not None and estimator.preprocessing != (0, 1))
-        ) or (
-            hasattr(estimator, "preprocessing_defences")
-            and (estimator.preprocessing_defences is not None and estimator.preprocessing_defences != [])
-        ):
+        if not estimator.all_framework_preprocessing:
```

Comment:
Line 88-91: I think the if condition and the message in the raise are not related. Should we change the message to update the meaning of the condition?

Reply:
I agree, fixed with 53baf291bd0fa5749bffeac752b4d77a1ee5ceea